assets/js/modules/pagespeed-insights/datastore/report.js

```diff
@@ -0,0 +1,228 @@
+/**
+ * modules/pagespeed-insights data store: report.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import { isPlainObject } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { isURL } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+
+// Actions
+const FETCH_REPORT = 'FETCH_REPORT';
+const START_FETCH_REPORT = 'START_FETCH_REPORT';
+const FINISH_FETCH_REPORT = 'FINISH_FETCH_REPORT';
+const CATCH_FETCH_REPORT = 'CATCH_FETCH_REPORT';
+const RECEIVE_REPORT = 'RECEIVE_REPORT';
+
+export const INITIAL_STATE = {
+	isFetchingReport: {},
+	reports: {},
+};
+
+export const actions = {
+	/**
+	 * Fetches a PageSpeed Insights report.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {string} url      URL used for generating the report.
+	 * @param {string} strategy Strategy used for generating the report.
+	 * @return {Object} Object with {response, error}
+	 */
+	*fetchReport( url, strategy ) {
+		invariant( isURL( url ), 'a valid url is required to fetch a report.' );
+		invariant( typeof strategy === 'string', 'a valid strategy is required to fetch a report.' );
+		let response, error;
+
+		yield {
+			payload: { strategy, url },
+			type: START_FETCH_REPORT,
+		};
+
+		try {
+			const report = yield {
```

Comment:
Looks like this code isn't covered by a test for its response, because this constant is being created here, sent to the `receiveReport` action, but then never used again. From the structure of this action, it looks like this should be assigned to the `response` variable you define above and return below.

```suggestion
			response = yield {
```

Reply:
:+1: